components/fxa-client/src/migrator.rs

```diff
@@ -68,10 +100,12 @@ impl FirefoxAccount {
             k: k_sync,
             kid,
         };
-        self.state.session_token = Some(migration_session_token);
+        self.state.session_token = Some(migration_session_token.to_string());
         self.state
             .scoped_keys
             .insert(scopes::OLD_SYNC.to_string(), k_sync_scoped_key);
+        // clear the migration state, we are done.
+        self.state.in_flight_migration = None;
```

Comment:
Nit: Can we move this assignment to line 43, so the reader can tell right away that `in_flight_migration` is only ephemeral in the optimistic case?

Reply:
Actually no, because this method gets called somewhere else, sorry.